FirebaseTestProject/Views/NewPostForm.swift

```diff
@@ -10,31 +10,73 @@ import SwiftUI
 struct NewPostForm: View {
     @State private var title = ""
     @State private var postContent = ""
+    @State private var imageSourceType: ImagePickerView.SourceType?
+    @State private var image: UIImage?
+    
     @FocusState private var showingKeyboard: Bool
+    
     @StateObject private var submitTask = TaskViewModel()
+    
     @Environment(\.user) private var user
+
     
     var body: some View {
         Form {
             TextField("Title", text: $title)
                 .focused($showingKeyboard)
-            TextField("Post content", text: $postContent)
+            TextEditor(text: $postContent)
                 .focused($showingKeyboard)
+                .multilineTextAlignment(/*@START_MENU_TOKEN@*/.leading/*@END_MENU_TOKEN@*/)
+                .frame(width: 300, height: 300, alignment: .topLeading)
+                
+            if let image = image {
+                Image(uiImage: image)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 200, height: 200, alignment: .center)
+            } else {
+                Image(systemName: "photo")
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 200, height: 200, alignment: .center)
+            }
+            VStack {
+                HStack(spacing: 30) {
+                    Text("Attach Image")
+                    Spacer()
+                    Button {
+                        imageSourceType = .photoLibrary
+                    }  label: {
+                        Label("Choose from Library", systemImage: "photo")
+                    }
+                    Button {
+                        imageSourceType = .camera
+                    }  label: {
+                        Label("Take Photo", systemImage: "camera")
+                    }
+                }
+                .buttonStyle(.borderless)
+                .labelStyle(.iconOnly)
+            }
             Button("Submit", action: submitPost)
         }
         .alert("Cannot Submit Post", isPresented: $submitTask.isError, presenting: submitTask.error) { error in
```

Comment:
Does `isError` ever reset in the ViewModel?  Not part of this PR, just curious.

Reply:
It resets, but not within the view model. Since it’s passed as an `isPresented` binding, it gets updated automatically when the alert is dismissed.